### PR TITLE
Update tests and renaming

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Follow-up PR to implement the changes requested by @gregory-schiano at #5 

- Update logic and add try-except for maxconn data validation
- add unit test for maxconn validation
- Update naming of docstrings to `"reload"`

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by mthaddon — APPROVED** [review]
LGTM

File: src/haproxy.py [hunk 1]
```diff
@@ -106,5 +106,13 @@ def _render_haproxy_config(self, config: CharmConfig) -> None:
         self._render_file(HAPROXY_CONFIG, rendered, 0o644)
 
     def _reload_haproxy_service(self) -> None:
-        """Restart the haporxy service."""
-        systemd.service_reload(HAPROXY_SERVICE)
+        """Reload the haporxy service.
```
Comment (alithethird) [inline src/haproxy.py:109]: Nit pick:
`s/haporxy/haproxy`

**Review by alithethird — APPROVED** [review]
Just left a nitpick, other wise LGTM :rocket:

File: src/haproxy.py (hunk 1 above)
Comment (Thanhphan1147) [reply src/haproxy.py:109]: Thanks! will update that in the next PR.

**Pull request merged**